swim/disseminator.go

```diff
@@ -245,3 +248,60 @@ func (d *disseminator) ChangesCount() int {
 	d.RUnlock()
 	return c
 }
+
+// tryStartReverseFullSync fires a goroutine that performs a full sync. We omit
+// the reverse full sync if the maximum number of processes are already
+// running. This ensures no more than reverseFullSyncJobs processes are
+// running concurrently.
+func (d *disseminator) tryStartReverseFullSync(target string, timeout time.Duration) {
+	if d.reverseFullSyncJobs == nil {
+		d.reverseFullSyncJobs = make(chan struct{}, d.node.maxReverseFullSyncJobs)
+	}
```

Comment:
Yeah I wasn't sure, I liked that all the code was in the same place, but I think you're right.